adding livekit docs

gkarthi-signoz opened the pull request.

gkarthi-signoz committed:
adding livekit docs

gkarthi-signoz committed:
Update livekit-observability.mdx

gkarthi-signoz committed:
Update livekit-observability.mdx

gkarthi-signoz committed:
updating dashboard image

gkarthi-signoz commented on data/docs/livekit-observability.mdx:
```diff
@@ -0,0 +1,582 @@
+
+## Overview
+
+This guide walks you through setting up observability and monitoring for LiveKit using [OpenTelemetry](https://opentelemetry.io/) and exporting logs, traces, and metrics to SigNoz. With this integration, you can observe various models performance, capture request/response details, and track system-level metrics in SigNoz, giving you real-time visibility into latency, error rates, and usage trends ...
```
done

gkarthi-signoz commented:
@claude /review

gkarthi-signoz committed:
Document OTLP environment variables for instrumentation

Added environment variable descriptions for OTLP configuration.

gkarthi-signoz commented on data/docs/livekit-observability.mdx:
```diff
@@ -0,0 +1,581 @@
+OTEL_LOGS_EXPORTER=otlp \
+OTEL_PYTHON_LOG_CORRELATION=true \
+OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED=true \
+<your_run_command with opentelemetry-instrument>
```
done

gkarthi-signoz committed:
Fix grammar in LiveKit observability guide

Corrected grammar by adding 'the' before 'performance' in the overview section.

makeavish commented on data/docs/livekit-observability.mdx:
```diff
@@ -0,0 +1,591 @@
+
+## Prerequisites
+
+- A [SigNoz Cloud account](https://signoz.io/teams/) with an active ingestion key
```
Mention Self Hosted SigNoz as well

makeavish commented on data/docs/livekit-observability.mdx:
```diff
@@ -0,0 +1,591 @@
+
+Instrumenting LiveKit in your AI applications with telemetry ensures full observability across your voice agent workflows, making it easier to debug issues, optimize performance, and understand user interactions. By leveraging SigNoz, you can analyze correlated traces, logs, and metrics in unified dashboards, configure alerts, and gain actionable insights to continuously improve reliability, resp ...
+
+## Prerequisites
```
Add this drop-in snippet before prerequisites - https://github.com/SigNoz/signoz.io/blob/main/CONTRIBUTING.md?plain=1#L252

makeavish commented on data/docs/livekit-observability.mdx:
```diff
@@ -0,0 +1,591 @@
+    def _on_metrics_collected(ev: MetricsCollectedEvent):
+        metrics.log_metrics(ev.metrics)
+
+    # To use a realtime model instead of a voice pipeline, use the following session setup instead.
```
Is the commented code needed for setting up instrumentation?

gkarthi-signoz committed:
Revise LiveKit observability documentation

Updated headings and added links for clarity in the LiveKit observability documentation.

gkarthi-signoz committed:
Remove OTEL environment variable descriptions

Removed detailed descriptions of various OTEL environment variables.

gkarthi-signoz committed:
Merge branch 'main' into livekit-docs
